src/C++/SSLSocketConnection.cpp

```diff
@@ -182,52 +184,62 @@ bool SSLSocketConnection::processQueue()
     return false;
     
   const std::string& msg = m_sendQueue.front();
-  m_sendLength = 0;
-
-  while (m_sendLength < (int)msg.length())
+  
```

Comment:
This method has been re-written to be closer to the non-ssl version of socket connection. It will no longer spin-loop until the entire message is sent. It will rather send as much as it can, and keep track of m_sendLength to resume when it's re-invoked